logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -170,28 +173,30 @@ def initialize(queue, size, wait)
         @size = size
         @wait = wait
 
-        @originals = Hash.new
-
         # TODO: disabled for https://github.com/elastic/logstash/issues/6055 - will have to properly refactor
         # @cancelled = Hash.new
 
+        #Sizing HashSet to size/load_factor to ensure no rehashing
+        @originals = java.util.HashSet.new(size * 4 / 3 + 1, 0.75)
```

Comment:
@andrewvc I benchmarked with `1` and `0.75` and saw no measurable difference in GC times or even bytes/objects allocated => I decided to be nice to the CPU given that we do quite a few `includes` checks on this thing.